src/test/java/ibm/jceplus/junit/base/BaseTestMessageDigestClone.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright IBM Corp. 2024
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution.
+ */
+
+package ibm.jceplus.junit.base;
+
+import java.security.MessageDigest;
+import java.util.Arrays;
+
+public class BaseTestMessageDigestClone extends BaseTest {
+    final byte[] input_1 = {(byte) 0x61, (byte) 0x61, (byte) 0x61, (byte) 0x61, (byte) 0x61,
+            (byte) 0x61, (byte) 0x61, (byte) 0x61, (byte) 0x61, (byte) 0x61};
+
+    final byte[] input_2 = {(byte) 0x61, (byte) 0x62, (byte) 0x63};
+
+    final byte[] input_3 = {(byte) 0x61, (byte) 0x62, (byte) 0x63, (byte) 0x64, (byte) 0x62,
+            (byte) 0x63, (byte) 0x64, (byte) 0x65, (byte) 0x63, (byte) 0x64, (byte) 0x65,
+            (byte) 0x66, (byte) 0x64, (byte) 0x65, (byte) 0x66, (byte) 0x67, (byte) 0x65,
+            (byte) 0x66, (byte) 0x67, (byte) 0x68, (byte) 0x66, (byte) 0x67, (byte) 0x68,
+            (byte) 0x69, (byte) 0x67, (byte) 0x68, (byte) 0x69, (byte) 0x6a, (byte) 0x68,
+            (byte) 0x69, (byte) 0x6a, (byte) 0x6b, (byte) 0x69, (byte) 0x6a, (byte) 0x6b,
+            (byte) 0x6c, (byte) 0x6a, (byte) 0x6b, (byte) 0x6c, (byte) 0x6d, (byte) 0x6b,
+            (byte) 0x6c, (byte) 0x6d, (byte) 0x6e, (byte) 0x6c, (byte) 0x6d, (byte) 0x6e,
+            (byte) 0x6f, (byte) 0x6d, (byte) 0x6e, (byte) 0x6f, (byte) 0x70, (byte) 0x6e,
+            (byte) 0x6f, (byte) 0x70, (byte) 0x71};
+
+    public BaseTestMessageDigestClone(String providerName) {
+        super(providerName);
+    }
+
+    public void testUpdateCloneSameUpdate() throws Exception {
+        MessageDigest md = MessageDigest.getInstance("SHA-256", providerName);
+
+        md.update(input_1);
+
+        MessageDigest mdCopy = (MessageDigest) md.clone();
+
+        byte[] digest1 = md.digest(input_2);
+        byte[] digest2 = mdCopy.digest(input_2);
+
+        assertTrue("Digest of original did not match clone's digest", Arrays.equals(digest1, digest2));
```

Comment:
Instead of `assertTrue` and `assertFalse` can we use `assertArrayEquals("Digest of original did not match clone's digest",digest1, digest2)` when possible? Would prefer these as a habit as it gives more information on failure and clues as to how arrays are different.

Reply:
I substituted `assertTrue` with suggestion. I couldn't find anything similar for the `assertFalse` cases.

Reply:
I couldnt find the reverse of `assertArrayEquals` either so using `assertFalse` seems ok.